servers/src/main/java/tachyon/master/file/FileSystemMaster.java

```diff
@@ -439,16 +366,14 @@ public void completeFile(long fileId)
       InodeFile fileInode = (InodeFile) inode;
       List<Long> blockIdList = fileInode.getBlockIds();
       List<BlockInfo> blockInfoList = mBlockMaster.getBlockInfoList(blockIdList);
-      if (blockInfoList.size() != blockIdList.size()) {
-        throw new BlockInfoException("Cannot complete file without all the blocks committed");
-      }
 
-      // Verify that all the blocks (except the last one) is the same size as the file block size.
-      long fileLength = 0;
+      // Iterate from all in-memory file blocks, computing the length and verify that all the blocks
```

Comment:
Technically, this is not iterating over in-memory blocks only, but all blocks known to Tachyon.